source/common/common/utility.h

```diff
@@ -185,10 +185,30 @@ class StringUtil {
    * @param trim_whitespace remove leading and trailing whitespaces from each of the split
    * string views; default = true.
    * @return sub-string of the string view if any.
+   *
+   * E.g.,
+   *
+   * cropRight("foo ; ; ; ; ; ; ", "b", true) == "foo"
```

Comment:
OK I actually recommend removing the trim_whitespace optional arg.  I think we have trimming & cropping done independently if needed, and anyway you were setting it to false in all cases, so maybe just simplify the utility function so it just does what you are using.  That'd save some conditionals in the request path.

Reply:
SGTM 👍